client/src/components/VideoForm.js

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+const VideoForm = ({ getAllVideos }) => {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [backendMessage, setBackendMessage] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const addHandler = async (e) => {
+    e.preventDefault();
+    if (!title.trim() || !url.trim()) {
+      setErrorMessage("Title and URL are required.");
+      return;
+    }
+    const youtubeRegExp =
+      /^(https?:\/\/)?(www\.)?(youtube\.com|youtu\.?be)\/.+$/;
+    if (!youtubeRegExp.test(url)) {
+      setErrorMessage("Please enter a valid YouTube URL.");
+      return;
+    }
+
+    // Perform the API call
+    const newVideo = {
+      title,
+      url,
+      ratingup: 0,
+      ratingdown: 0,
+      date: new Date().getTime(),
+    };
+    try {
+      const response = await fetch(`${process.env.REACT_APP_SERVERURL}/`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(newVideo),
+      });
+      if (response.status === 400) {
+        const errorData = await response.json();
+        const validationErrors = errorData.errors;
+        setErrorMessage(validationErrors.map((error) => error.msg).join(", "));
+      } else if (response.status === 409) {
+        const existingError = await response.json();
+        setErrorMessage(existingError.message);
+      } else if (response.status !== 201) {
```

Comment:
Given you've been doing lots of `===` checks before, I'd more expect you to handle the `=== 201` case here, and for the else to be "all other status codes", rather than the other way around. Your core here works, but it's kind of surprising, which makes it slightly harder to read and work with. People are really good at spotting patterns, so breaking that pattern can be confusing.